networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/GossipFailureLogger.java

```diff
@@ -25,48 +26,63 @@
 public class GossipFailureLogger {
   private static final Logger LOG = LogManager.getLogger();
 
+  private final boolean shouldSuppress;
   private final String messageType;
-  private UInt64 lastErroredSlot;
+  private Optional<UInt64> lastErroredSlot;
   private Throwable lastRootCause;
 
-  public GossipFailureLogger(final String messageType) {
-    this.messageType = messageType;
+  public GossipFailureLogger(final String messageType, final boolean shouldSuppress) {
```

Comment:
I'd make this private and make 2 static creators with and without supression instead for greater clarity.